buildkite/resource_cluster.go

```diff
@@ -128,8 +128,18 @@ func (c *clusterResource) Read(ctx context.Context, req resource.ReadRequest, re
 		return
 	}
 
-	updateClusterResourceState(r.Organization.Cluster, &state)
-	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	if clusterNode, ok := r.GetNode().(*getNodeNodeCluster); ok {
+		if clusterNode == nil {
+			resp.Diagnostics.AddError(
+				"Unable to get Cluster",
+				"Error getting Cluster: nil response",
+			)
+			return
+		}
+		updateClusterResourceState(&state, *clusterNode)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	}
```

Comment:
We need an else here for if the cluster is not found. It needs to be removed from state. You can see similar in https://github.com/buildkite/terraform-provider-buildkite/blob/main/buildkite/resource_pipeline.go#L327-L331